packages/typescript-estree/src/parseSettings/ExpiringCache.ts

```diff
@@ -0,0 +1,111 @@
+export const DEFAULT_TSCONFIG_CACHE_DURATION_SECONDS = 30;
+
+/**
+ * A map with key-level expiration.
+ */
+export class ExpiringCache<TKey, TValue> {
+  readonly #cacheDurationSeconds: number | 'Infinity';
```

Comment:
I chose seconds here just because there's really no need for millisecond precision.
You're never going to want less than 1s cache duration, and nor are you going to want 30.5s cache duration.

Also it aligns well with the `process.hrtime` API - where the first value returned from the method is the number of seconds (and the second is the number of nanoseconds).